File: github-mvi-android/app/src/debug/java/com/rightpoint/mvi/example/injection/BuildTypeModule.java
package com.rightpoint.mvi.example.injection;

import android.os.StrictMode;

import dagger.Module;
import dagger.Provides;
import timber.log.Timber;

@Module
abstract class BuildTypeModule {
    @Provides
    static StrictMode.ThreadPolicy providesThreadPolicy() {
        return new StrictMode.ThreadPolicy.Builder()
                .detectAll()
                .penaltyLog()
                .build();
    }

    @Provides
    static StrictMode.VmPolicy providesVmPolicy() {
        return new StrictMode.VmPolicy.Builder()
                .detectAll()
                .penaltyLog()
                .build();
    }

    @Provides
    static Timber.Tree providesTree() {
        return new Timber.DebugTree();
    }
}

File: {{ cookiecutter.repo_name }}/app/src/debug/java/{{ cookiecutter.package_dir }}/BuildTypeModule.kt
package {{ cookiecutter.package_name }};

import android.os.StrictMode

import dagger.Module
import dagger.Provides
import timber.log.Timber

@Module
internal object BuildTypeModule {
    @Provides
    @JvmStatic
    fun providesThreadPolicy(): StrictMode.ThreadPolicy {
        return StrictMode.ThreadPolicy.Builder()
            .detectAll()
            .penaltyLog()
            .build()
    }

    @Provides
    @JvmStatic
    fun providesVmPolicy(): StrictMode.VmPolicy {
        return StrictMode.VmPolicy.Builder()
            .detectAll()
            .penaltyLog()
            .build()
    }

    @Provides
    @JvmStatic
    fun providesTree(): Timber.Tree {
        return Timber.DebugTree()
    }
}